beastify/popup/choose_beast.js

```diff
@@ -16,33 +16,39 @@ function beastNameToURL(beastName) {
 Listen for clicks in the popup.
 
 If the click is on one of the beasts:
-  Inject the "beastify.js" content script in the active tab.
+  (1) inject the beastify.js content script.
 
-  Then get the active tab and send "beastify.js" a message
-  containing the URL to the chosen beast's image.
+  (2) send a message to "beastify.js" that contains the URL to
+  the chosen beast's image.
 
-If it's on a button wich contains class "clear":
+If it's on a button which contains class "clear":
   Reload the page.
   Close the popup. This is needed, as the content script malfunctions after page reloads.
 */
 document.addEventListener("click", (e) => {
-  if (e.target.classList.contains("beast")) {
+
+  function messageBeastify() {
     var chosenBeast = e.target.textContent;
     var chosenBeastURL = beastNameToURL(chosenBeast);
-
-    browser.tabs.executeScript(null, {
-      file: "/content_scripts/beastify.js"
-    });
-
     var gettingActiveTab = browser.tabs.query({active: true, currentWindow: true});
     gettingActiveTab.then((tabs) => {
       browser.tabs.sendMessage(tabs[0].id, {beastURL: chosenBeastURL});
     });
   }
+
+  function reportError(error) {
+    console.error(`Could not inject content script: ${error}`);
+  }
+
+  if (e.target.classList.contains("beast")) {
+    browser.tabs.executeScript({
```

Comment:
In this specific example, the current window and tab are not expected to change because the script is running from an extension popup. However, this coding pattern should not be used in a background page, because there is no guarantee that the "current tab" at the time of invoking `browser.tabs.executeScript` matches the "active tab in the current window" (lines 33-35) at the time of sending the message. This situation occurs when the user switches tabs or windows between the two calls.

To have this guarantee, call `browser.tabs.query` in advance and use the retrieved `tabId` for both `sendMessage` and `executeScript`.

You can either implement this suggestion, or add a note in the comments to warn the developer about this.